From f0114fdd82d2712b9a6f163235a154eb003d2946 Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Thu, 11 Apr 2024 08:29:56 -0500
Subject: [PATCH 2/2] CMake: Locate and use the GL include directory

Find the GL include directory and use it in target_include_directories calls.
---
 CMake/options.cmake | 1 +
 src/CMakeLists.txt  | 3 +++
 2 files changed, 4 insertions(+)

diff --git a/CMake/options.cmake b/CMake/options.cmake
index c4ba98914..fb7ce5653 100644
--- a/CMake/options.cmake
+++ b/CMake/options.cmake
@@ -618,6 +618,7 @@ if(FLTK_BUILD_GL)
     # FIXME: we should find a better way to resolve this issue:
     # with GL, must use XQuartz libX11 else "Insufficient GL support"
     set(OPENGL_LIBRARIES -L${PATH_TO_GLULIB} -L${PATH_TO_GLLIB} -lX11 -lGLU -lGL)
+    find_path(OPENGL_INCLUDE_DIR NAMES GL/gl.h OpenGL/gl.h HINTS ${X11_INCLUDE_DIR})
     unset(HAVE_GL_GLU_H CACHE)
     find_file(HAVE_GL_GLU_H GL/glu.h PATHS ${X11_INCLUDE_DIR})
   else()
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 91a8c614c..b12880c3b 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -813,6 +813,7 @@ endif()
 if(FLTK_USE_GL)
   fl_add_library(fltk_gl STATIC "${GLCPPFILES};${GL_HEADER_FILES};${GL_DRIVER_HEADER_FILES}")
   target_link_libraries(fltk_gl PUBLIC ${OPENGL_LIBRARIES} fltk::fltk)
+  target_include_directories(fltk_gl PUBLIC ${OPENGL_INCLUDE_DIR} ${OPENGL_INCLUDE_DIRS})
   if(OPENGL_GLU_INCLUDE_DIR)
     target_include_directories(fltk_gl PUBLIC ${OPENGL_GLU_INCLUDE_DIR})
   endif()
@@ -882,6 +883,7 @@ if(FLTK_BUILD_SHARED_LIBS AND NOT MSVC)
   if(FLTK_USE_GL)
     fl_add_library(fltk_gl SHARED "${GLCPPFILES};${GL_HEADER_FILES};${GL_DRIVER_HEADER_FILES}")
     target_link_libraries(fltk_gl-shared PUBLIC ${OPENGL_LIBRARIES} fltk::fltk-shared)
+    target_include_directories(fltk_gl-shared PUBLIC ${OPENGL_INCLUDE_DIR} ${OPENGL_INCLUDE_DIRS})
     if(OPENGL_GLU_INCLUDE_DIR)
       target_include_directories(fltk_gl-shared PUBLIC ${OPENGL_GLU_INCLUDE_DIR})
     endif()
@@ -924,6 +926,7 @@ if(FLTK_BUILD_SHARED_LIBS AND MSVC)
 
   if(OPENGL_FOUND)
     target_link_libraries(fltk-shared PUBLIC ${OPENGL_LIBRARIES})
+    target_include_directories(fltk_gl-shared PUBLIC ${OPENGL_INCLUDE_DIR} ${OPENGL_INCLUDE_DIRS})
     if(OPENGL_GLU_INCLUDE_DIR)
       target_include_directories(fltk-shared PUBLIC ${OPENGL_GLU_INCLUDE_DIR})
     endif()
-- 
2.44.0

